certificates/certutils.go

```diff
@@ -45,15 +45,15 @@ func ScrapeRootCertificatesFromURL(URL string) (*x509.Certificate, error) {
 		return nil, err
 	}
 
-	peerCertificates := conn.ConnectionState().PeerCertificates
-	if len(peerCertificates) == 0 {
-		err = fmt.Errorf("no peer certificates found at %s", URL)
-		logrus.Error(err)
-		return nil, err
+	chains := conn.ConnectionState().VerifiedChains
+	if len(chains) == 0 {
+		return nil, fmt.Errorf("no certificates found at %s", URL)
 	}
-
-	rootCertificate := peerCertificates[len(peerCertificates)-1]
-	return rootCertificate, nil
+	rootCertificate := chains[len(chains)-1]
```

Comment:
from the docs:

> `VerifiedChains` is a **list** of **one or more chains** where the first element is
>	`PeerCertificates[0]` and the last element is from `Config.RootCAs`.
>	
>	It's set if `Config.InsecureSkipVerify` is `false`.
>	`VerifiedChains` and its contents should not be modified.

```go
VerifiedChains [][]*x509.Certificate
```

So maybe we can simply pick the first chain instead of the last one?

```suggestion
	rootCertificate := chains[0]
```

Reply:
I wouldn't name `rootCertificate` the `chains[0]`. Probably `firtsChain` is a better name

Reply:
@cmaglie We found an edge case in which the unor4 doesn't follow exactly the same nina behavior of mbedtls. This means that some root certificates won't work. The quick fix for this is to always put all the root certificates.
I've tested on mkrwifi1010, uno r4, portenca c33 and all seems working fine.